File: gamelogic/src/main/java/com/alta_v2/game/gamelogic/GameLogic.java
package com.alta_v2.game.gamelogic;

import com.alta_v2.game.AltaV2;
import com.alta_v2.mediator.serde.ActionListener;
import lombok.Getter;

import javax.inject.Inject;

public class GameLogic {

    @Getter
    private final AltaV2 altaV2;

    @Inject
    public GameLogic(ActionListener actionListener, AltaV2 altaV2) {
        this.altaV2 = altaV2;
        this.altaV2.setInputListener(actionListener);
    }

}

File: gamelogic/src/main/kotlin/com/alta_v2/game/gamelogic/GameLogic.kt
package com.alta_v2.game.gamelogic

import com.alta_v2.game.AltaV2
import com.alta_v2.mediator.serde.ActionListener
import com.google.inject.Inject

class GameLogic @Inject constructor(val altaV2: AltaV2, actionListener: ActionListener) {

    init {
        altaV2.setInputListener(actionListener)
    }

}